### PR TITLE
Simprocedure fix

### PR DESCRIPTION
**Pull request by Airtnp**

### CODE REVIEW

File: angr/procedures/linux_kernel/mprotect.py [hunk 1]
```diff
@@ -8,19 +8,22 @@ class mprotect(angr.SimProcedure):
...
-            addr = self.state.solver.eval_one(addr)
+            try:
+                addr = self.state.solver.eval_one(addr)
+            except angr.errors.SimValueError:
+                raise angr.errors.SimValueError("mprotect can't handle symbolic addr")
+
+            try:
+                length = self.state.solver.eval_one(length)
+            except angr.errors.SimValueError:
+                raise angr.errors.SimValueError("mprotect can't handle symbolic length")
```
Comment (Mic92) [inline angr/procedures/linux_kernel/mprotect.py:19]: Why does it raise an exception here, when it is catched below again?

File: angr/procedures/libc/strstr.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def run(self, haystack_addr, needle_addr, haystack_strlen=None, needle_strlen=No
                     break
 
             cases.append([ self.state.solver.And(*exclusions), self.state.solver.BVV(0, self.state.arch.bits) ])
-            l.debug("... created %d cases", len(cases))
+            l.debug("... created %d cases", len(cases))y
```
Comment (Mic92) [inline angr/procedures/libc/strstr.py:59]: This should not be there.

Comment (Airtnp) [reply angr/procedures/libc/strstr.py:59]: changed in next commit

File: angr/procedures/linux_kernel/mprotect.py (hunk 1 above)
Comment (Airtnp) [reply angr/procedures/linux_kernel/mprotect.py:19]: the raise is original  code and catch it as a fallback

File: angr/procedures/libc/feof.py [hunk 3]
```diff
@@ -15,5 +15,5 @@ def run(self, file_ptr):
         fileno = self.state.mem[file_ptr + fd_offset:].int.concrete
         simfd = self.state.posix.get_fd(fileno)
         if simfd is None:
-            return None
+            return self.state.solver.If('feof', self.state.solver.BVV(1, self.state.arch.bits), 0)
```
Comment (Mic92) [inline angr/procedures/libc/feof.py:18]: Why this is an `If` here? The string should always evaluate to True.

Comment (Airtnp) [reply angr/procedures/libc/feof.py:18]: oh that‘s a mistake as it should create a new symbolic variable instead of string

Comment (Mic92) [reply angr/procedures/libc/feof.py:18]: If one pass NULL to feof should it not crash instead?

Comment (Airtnp) [reply angr/procedures/libc/feof.py:18]: changed in latest commit

**Pull request merged**